src/lib/media/manip.ts

```diff
@@ -168,26 +184,40 @@ async function doResize(localUri: string, opts: DoResizeOpts): Promise<Image> {
         width: resizeRes.width,
         height: resizeRes.height,
       }
+    } else {
+      try {
+        deleteAsync(resizeRes.path)
+      } catch (e) {
+        console.error(e)
+      }
     }
   }
   throw new Error(
     `This image is too big! We couldn't compress it down to ${opts.maxSize} bytes`,
   )
 }
 
-async function moveToPermanentPath(path: string, ext = ''): Promise<string> {
+async function moveToPermanentPath(path: string, ext = 'jpg'): Promise<string> {
   /*
   Since this package stores images in a temp directory, we need to move the file to a permanent location.
   Relevant: IOS bug when trying to open a second time:
   https://github.com/ivpusic/react-native-image-crop-picker/issues/1199
   */
   const filename = uuid.v4()
 
-  const destinationPath = joinPath(
-    RNFS.TemporaryDirectoryPath,
-    `${filename}${ext}`,
-  )
-  await RNFS.moveFile(path, destinationPath)
+  const destinationPath = joinPath(cacheDirectory ?? '', `${filename}.${ext}`)
```

Comment:
`cacheDirectory` will not be null _unless_ we call this on web. This function only ever gets called on native though, so we know that `cacheDirectory` will always be available.

Reply:
If `cacheDirectory` can never be null, let's assert that? `?? ''` isn't good cause it obscures intent for future readers